### PR TITLE
feat(mailer): add tests for service

### LINKED ISSUE #65: tests: write tests for mailer service
We have already set up tests for user service.

Take it as a reference, and write tests for mailer service.

Also, please attach screenshots of tests passing when sending PRs.

### PR DESCRIPTION
**Pull request by MASTERCHIEF011**
Fixes #65 
Screen Shot of test
image

### CODE REVIEW

**Review by YashKumarVerma — APPROVED** [review]
add screenshots of test passing

File: src/mailer/mailer.service.spec.ts [hunk 1]
```diff
@@ -1,18 +1,81 @@
+import { JwtModule } from '@nestjs/jwt';
+import { getModelToken, MongooseModule } from '@nestjs/mongoose';
 import { Test, TestingModule } from '@nestjs/testing';
+import { Model } from 'mongoose';
+import { User, UserDocument, UserSchema } from 'src/user/user.schema';
```
Comment (YashKumarVerma) [inline src/mailer/mailer.service.spec.ts:5]: use relative imports else jest won't be able to resolve the location

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]

File: src/mailer/mailer.service.spec.ts [hunk 2]
```diff
@@ -1,18 +1,85 @@
...
+import { UserService } from '../user/user.service';
+
+
+// const mockUser = (mock?: Partial<User>): Partial<UserDocument> => ({
+//   name: mock.name || 'some user',
+//   batch: mock.batch || '19',
+//   branch: mock.branch || 'BCE',
+//   personalEmail: mock.personalEmail || 'someone@example.com',
+//   collegeEmail: mock.collegeEmail || 'someoe@edu.in',
+// });
```
Comment (YashKumarVerma) [inline src/mailer/mailer.service.spec.ts:19]: delete these as not needed.

File: src/mailer/mailer.service.spec.ts [hunk 3]
```diff
@@ -1,18 +1,85 @@
...
+        }),
+      ],
+      providers: [MailerService,UserService,
+      //   {
+      //   provide: getModelToken(User.name),
+      //   useValue: {
+      //     findByIdAndUpdate: jest.fn().mockResolvedValue(mockUser()),
+      //   },
+      // },
+    ],
```
Comment (YashKumarVerma) [inline src/mailer/mailer.service.spec.ts:44]: why commented ?

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]

File: src/mailer/mailer.service.spec.ts (hunk 3 above)
Comment (MASTERCHIEF011) [reply src/mailer/mailer.service.spec.ts:44]: line no 41: findByIdAndUpdate is a function that has been used in mailer.service.ts but has not been tested here.

**Review by YashKumarVerma — APPROVED** [review]

**Comment by YashKumarVerma** [comment]
Thanks !

**Pull request merged**